integration_tests/contracts/contracts/TestICA.sol

```diff
@@ -76,39 +80,37 @@ contract TestICA {
 
     function callSubmitMsgs(string memory connectionID, bytes memory data, uint256 timeout) public returns (uint64) {
         require(account == msg.sender, "not authorized");
-        lastAckSeq = ica.submitMsgs(connectionID, data, timeout);
-        return lastAckSeq;
+        lastSeq = ica.submitMsgs(connectionID, data, timeout);
+        return lastSeq;
     }
 
     function delegateSubmitMsgs(string memory connectionID, bytes memory data, uint256 timeout) public returns (uint64) {
         (bool result, bytes memory data) = icaContract.delegatecall(encodeSubmitMsgs(connectionID, data, timeout));
         require(result, "call failed");
-        lastAckSeq = abi.decode(data, (uint64));
-        return lastAckSeq;
+        lastSeq = abi.decode(data, (uint64));
+        return lastSeq;
     }
 
     function staticSubmitMsgs(string memory connectionID, bytes memory data, uint256 timeout) public returns (uint64) {
         (bool result, bytes memory data) = icaContract.staticcall(encodeSubmitMsgs(connectionID, data, timeout));
         require(result, "call failed");
-        lastAckSeq = abi.decode(data, (uint64));
-        return lastAckSeq;
-    }
-
-    function getLastAckSeq() public view returns (uint256) {
-        return lastAckSeq;
+        lastSeq = abi.decode(data, (uint64));
+        return lastSeq;
     }
 
-    function getLastAck() public view returns (bool) {
-        return lastAck;
+    function getLastSeq() public view returns (uint256) {
+        return lastSeq;
     }
 
     function onPacketResultCallback(uint64 seq, bool ack) external payable returns (bool) {
         // To prevent called by arbitrary user
         require(msg.sender == module_address);
-        lastAckSeq = seq;
-        lastAck = ack;
-        acknowledgement[seq] = ack;
-        emit OnPacketResult(seq, ack);
+        Status status = Status.TIMEOUT;
```

Comment:
timeout covers the failure case as well?

Reply:
I haven't get the err ack reproduced yet

Reply:
Try sending an invalid message